R/oauth-init.R

```diff
@@ -211,8 +211,12 @@ check_scope <- function(x) {
   paste(x, collapse = " ")
 }
 
-# Wrap base::interactive in a non-primitive function so that the call can be mocked for testing
-is_interactive <- function() interactive()
+# gargle needs to access (pseudo-)OOB flow from Google Colab, so we need to
+# be able to use the "rlang_interactive" option to signal that we are in
+# an interactive environment. httr does not have an rlang dependency.
+is_interactive <- function() {
+  getOption("rlang_interactive") %||% interactive()
+}
```

Comment:
I can't use `rlang::is_interactive()` because httr does not depend on rlang at this point. And it doesn't seem worth bringing a whole standalone file in for this.